service/worker/stream.go

```diff
@@ -80,32 +76,18 @@ func (ps *PushStream) Start() error {
 				}
 			default:
 				logger.Ctx(ps.ctx).Infow("worker: sending a subscriber pull request", "subscription", ps.subcriptionName, "subscriberId", ps.subs.GetID())
-				ps.subs.GetRequestChannel() <- &subscriber.PullRequest{10}
+				ps.subs.GetRequestChannel() <- &subscriber.PullRequest{MaxNumOfMessages: 10}
 				logger.Ctx(ps.ctx).Infow("worker: waiting for subscriber data", "subscription", ps.subcriptionName, "subscriberId", ps.subs.GetID())
-				res := <-ps.subs.GetResponseChannel()
-				logger.Ctx(ps.ctx).Infow("worker: writing subscriber data to channel", "res", res, "count", len(res.ReceivedMessages), "subscription", ps.subcriptionName, "subscriberId", ps.subs.GetID())
-				ps.responseChan <- res
-			}
-		}
-		logger.Ctx(ps.ctx).Infow("worker: returning from pull stream go routine", "subscription", ps.subcriptionName, "subscriberId", ps.subs.GetID())
-		return nil
-	})
-
-	errGrp.Go(func() error {
-		// read from response channel and fire a webhook
-		for {
-			select {
-			case <-gctx.Done():
-				return gctx.Err()
-			default:
-				data := <-ps.responseChan
+				// wait for message to be available on the response channel
+				data := <-ps.subs.GetResponseChannel()
```

Comment:
read it in a separate case statement  ?

Reply:
Currently `ps.subs.GetResponseChannel()` is a blocking call. Every request will wait for a response. If I move it to a case, then items in requests chan can exceed response chan. Are we ok with this?

Reply:
we can control no of of requests by limiting the request and response channel sizes. 
keeping request response in processing in one block will make it slow, we should then consider hitting webhooks in separate go routines